src/components/ui/sidebar/nav-main.tsx

```diff
@@ -30,15 +30,55 @@ import { Avatar } from "@/components/Common/Avatar";
 
 import { NavigationLink } from "./facility-nav";
 
-const isChildActive = (link: NavigationLink) => {
+/* Converts a route pattern like '/path/:id' to a RegExp to match dynamic segments in URLs.
+Helps in checking if the current path matches a pattern for active link detection. */
+const matchPath = (pattern: string, path: string | null) => {
+  const regex = new RegExp(`^${pattern.replace(/:\w+/g, "[^/]+")}`);
```

Comment:
```suggestion
  const regex = new RegExp(`^${pattern.replace(/:\w+/g, "[^/]+")}(?:/|$)`);
```

So that:
- /product
- /product_knowledge

don't match 👍 Should work, but be sure to test.

Reply:
it is working .updated the regex